code/src/components/Checkbox.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import { useState } from 'react';
+
+import styled from 'styled-components'
+
+export const CheckBox = ({ isChecked, onChange }) => {
+  const [checked, setChecked] = useState(isChecked);
+
+  const handleChange = event => {
+    onChange();
+    setChecked(event.target.checked);
+  };
+
+  //for the checkbox styling I checked an article on medium and adapted it to my code
+  const CheckboxContainer = styled.div`
+    display: inline-block;
+    vertical-align: middle;
+`
+
+  const Icon = styled.svg`
+    fill: none;
+    stroke: white;
+    stroke-width: 2px;
+`
+
+  const HiddenCheckbox = styled.input.attrs({ type: 'checkbox' })`
+    border: 0;
+    clip: rect(0 0 0 0);
+    clippath: inset(50%);
+    height: 1px;
+    margin: -1px;
+    overflow: hidden;
+    padding: 0;
+    position: absolute;
+    white-space: nowrap;
+    width: 1px;
+`
+
+  const StyledCheckbox = styled.div`
+    width: 20px;
+    height: 20px;
+    margin-top:5px;
+    background: ${props => (props.checked ? '#1ba341' : '#e6e6e6')};
+    border-radius: 3px;
+    transition: all 150ms;
+    cursor: pointer;
+
+    ${Icon} {
+      visibility: ${props => (props.checked ? 'visible' : 'hidden')}
+    };
+`
+
+  const Checkbox = ({ className, checked, ...props }) => (
+    <CheckboxContainer className={className}>
+      <HiddenCheckbox checked={checked} {...props} />
+      <StyledCheckbox checked={checked}>
+        <Icon viewBox="0 0 24 24">
+          <polyline points="20 6 9 17 4 12" />
+        </Icon>
+      </StyledCheckbox>
+    </CheckboxContainer>
+  );
+
+return( 
+  <div>
+    <label>
+      <Checkbox checked={checked} onChange={handleChange}></Checkbox>
+    </label>
+  </div>
+);
+};
```

Comment:
your costume checkbox looks great and works flawless!

Reply:
I agree that it looks great but there's a little bug appearing, if you check the first todo and then remove it - the second one will get checked. Take another look at this and you're good to go :)

Reply:
Oh my 😣 well spotted! It was my `Item key` 😓 the key was the index and what I think it was happening was when I removed one to do, the index changes and even though the complete state was false, the virtual DOM was just saying that todo with that index was "checked" and sending that info to the checkbox. Changed the item key to the item.id and the bud is solved 🥳